include/deal.II/fe/fe_interface_values.h

```diff
@@ -0,0 +1,689 @@
+  if (at_boundary())
+    Assert(fe_index == 0, ExcMessage("A boundary facet only has FE index 0."));
+
+  if (fe_index == 0)
+    {
+      Assert(fe_dof_index < n_dofs_fe, ExcMessage("invalid fe_dof_index"));
+      return fe_dof_index;
+    }
+  else if (fe_index == 1)
+    {
+      Assert(fe_dof_index < n_dofs_fe_neighbor,
+             ExcMessage("invalid fe_dof_index"));
+      return fe_dof_index + n_dofs_fe;
+    }
+
+  // return something invalid:
+  return numbers::invalid_unsigned_int;
+}
+
+
+
+template <int dim, int spacedim>
+const FEFaceValuesBase<dim, spacedim> &
+FEInterfaceValues<dim, spacedim>::get_fe_values(
+  const unsigned int cell_or_neighbor) const
+{
+  AssertIndexRange(cell_or_neighbor, 2);
+  return (cell_or_neighbor == 0) ? get_fe_values() : get_fe_values_neighbor();
+}
+
+
+
+template <int dim, int spacedim>
+const FEFaceValuesBase<dim, spacedim> &
+FEInterfaceValues<dim, spacedim>::get_fe_values() const
+{
+  return *fe_face_values;
+}
+
+
+
+template <int dim, int spacedim>
+const FEFaceValuesBase<dim, spacedim> &
+FEInterfaceValues<dim, spacedim>::get_fe_values_neighbor() const
+{
+  Assert(!at_boundary(), ExcMessage("Not possible for boundary Facet."));
+  return *fe_face_values_neighbor;
+}
+
+
+
+template <int dim, int spacedim>
+Tensor<1, spacedim>
+FEInterfaceValues<dim, spacedim>::normal(const unsigned int q_point_index) const
+{
+  return fe_face_values->normal_vector(q_point_index);
+}
+
+
+
+template <int dim, int spacedim>
+double
+FEInterfaceValues<dim, spacedim>::shape_value(
+  const bool         current_cell,
+  const unsigned int interface_dof_index,
+  const unsigned int q_point,
+  const unsigned int component) const
+{
+  const unsigned int shape_fct =
+    interface_dof_index_to_fe_dof_index(interface_dof_index);
+  const unsigned int fe_idx =
+    interface_dof_index_to_fe_index(interface_dof_index);
+
+  if (current_cell && fe_idx == 0)
+    return get_fe_values().shape_value_component(shape_fct, q_point, component);
+  if (!current_cell && fe_idx == 1)
+    return get_fe_values_neighbor().shape_value_component(shape_fct,
+                                                          q_point,
+                                                          component);
+
+  return 0.0;
+}
+
+
+
+template <int dim, int spacedim>
+double
+FEInterfaceValues<dim, spacedim>::jump(const unsigned int interface_dof_index,
+                                       const unsigned int q_point,
+                                       const unsigned int component) const
+{
+  const unsigned int shape_fct =
+    interface_dof_index_to_fe_dof_index(interface_dof_index);
+  const unsigned int fe_idx =
+    interface_dof_index_to_fe_index(interface_dof_index);
+
+  if (fe_idx == 0)
+    return get_fe_values().shape_value_component(shape_fct, q_point, component);
+  else
+    {
+      if (at_boundary())
+        return 0.0;
+      else
+        return -get_fe_values_neighbor().shape_value_component(shape_fct,
```

Comment:
same question here: what happens for continuous functions? Do you get a negative here and a positive from the other side, that are then added up?